src/site/antora/modules/ROOT/pages/manual/customconfig.adoc

```diff
@@ -14,372 +14,232 @@
+In this section we will share guides on programmatically configuring Log4j Core for certain use cases.
 
-1.  If the configuration file is changed the configuration will be
-reloaded and the manual changes will be lost.
-2.  Modification to the running configuration requires that all the
-methods being called (addAppender and addLogger) be synchronized.
+[#load-config-file]
+=== Loading a configuration file
 
-As such, the recommended approach for customizing a configuration is to
-extend one of the standard Configuration classes, override the setup
-method to first do super.setup() and then add the custom Appenders,
-Filters and LoggerConfigs to the configuration before it is registered
-for use.
+<<ConfigurationFactory>> provides the link:../javadoc/log4j-core/org/apache/logging/log4j/core/config/ConfigurationFactory.html#getInstance()[`getInstance()`] method returning a meta-`ConfigurationFactory` that combines the behaviour of all available ``ConfigurationFactory`` implementations, including xref:manual/configuration.adoc#configuration-factories[the predefined ones]; `XmlConfigurationFactory`, `JsonConfigurationFactory`, etc.
+You can use this `getInstance()` method to load a configuration file programmatically, granted that the input file format is supported by at least one of the available `ConfigurationFactory` plugins:
 
-The following example adds an Appender and a new LoggerConfig using that
-Appender to the current configuration.
-
-//TODO: update code example below with new plugin API
+.Snippet from an example {antora-examples-url}/manual/customconfig/Usage.java[`Usage.java`]
 [source,java]
 ----
-        final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
-        final Configuration config = ctx.getConfiguration();
-        Layout layout = PatternLayout.createDefaultLayout(config);
-        Appender appender = FileAppender.createAppender("target/test.log", "false", "false", "File", "true",
-            "false", "false", "4000", layout, null, "false", null, config);
-        appender.start();
-        config.addAppender(appender);
-        AppenderRef ref = AppenderRef.createAppenderRef("File", null, null);
-        AppenderRef[] refs = new AppenderRef[] {ref};
-        LoggerConfig loggerConfig = LoggerConfig.createLogger("false", "info", "org.apache.logging.log4j",
-            "true", refs, null, config, null );
-        loggerConfig.addAppender(appender, null, null);
-        config.addLogger("org.apache.logging.log4j", loggerConfig);
-        ctx.updateLoggers();
-}
+include::example$manual/customconfig/Usage.java[tag=loadConfigurationFile,indent=0]
 ----
+<1> Passing the optional `LoggerContext` argument as null, since this `Configuration` is not associated with a `LoggerContext` yet
+<2> Passing the optional configuration name argument as null, it will default to the configuration source location
+<3> URI pointing to the configuration file; `file://path/to/log4j2.xml`, `classpath:log4j2.xml`, etc.
```

Comment:
Maybe we should shorten the explanations:

- we can pass `LoggerContext` as `null`,
- we can pass `name` as `null`, since it is not used if `configLocation` is provided.

"since this `Configuration` is not associated with a `LoggerContext` yet" is IMHO wrong: each configuration is not associated with a logger context upon creation.

Reply:
Fixed in a44fe6cca642eb16fe263c049a55c3b6c0a72eaf.